### PR TITLE
[SecurityBundle] Fixed LogicException message of FirewallAwareTrait

### LINKED ISSUE #43145: [Security] FirewallAwareTrait throws Critical Error instead of throwing Exception
**Symfony version(s) affected**: 5.3.7

**Description**

The Symfony/Bundle/SecurityBundle/Security/FirewallAwareTrait throws Critical Error instead of throwing the right Exception with helpfull Message.
The Error thrown is "Call to undefined function Symfony\Bundle\SecurityBundle\LoginLink\LoginLinkHandler()".
The right Error message, I assume, would be "Cannot determine the correct Symfony\Bundle\SecurityBundle\LoginLink\LoginLinkHandler to use: there is no active Request and so, the firewall cannot be determined. Try using a specific Symfony\Bundle\SecurityBundle\LoginLink\Logi  
  nLinkHandler service."

**How to reproduce**

1. Create new Symfony project with php 7.4 `symfony new my_project_name --full
`
2. Configure login link like described [here](https://symfony.com/doc/current/security/login_link.html#configure-the-login-link-authenticator) 
3. Create new Command, inject LoginLinkHandlerInterface and try to call `$this->loginLinkHandler->createLoginLink($user);`
4. Execute Command and take a look at the Error message.

**Possible Solution**

Change Symfony/Bundle/SecurityBundle/Security/FirewallAwareTrait line to 
`throw new \LogicException('Cannot determine the correct '.$serviceIdentifier.' to use: there is no active Request and so, the firewall cannot be determined. Try using a specific '.$serviceIdentifier.' service.');` 

**Additional context**

### PR DESCRIPTION
**Pull request by fkropfhamer**
| Q             | A
| ------------- | ---
| Branch?       | 5.4 for features / 4.4 or 5.3 for bug fixes 
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Fix #43145 
| License       | MIT
| Doc PR        | symfony/symfony-docs#...

### CODE REVIEW

**Review by derrabus — APPROVED** [review]

**Comment by derrabus** [comment]
The change looks correct from my perspective. Would you be able to provide a failing test for your issue?

**Comment by fkropfhamer** [comment]
@derrabus Do you mean a unit test?

**Comment by derrabus** [comment]
Yes. A unit test that fails without and passes with your change.

**Comment by fkropfhamer** [comment]
Ok, I will try!

File: src/Symfony/Bundle/SecurityBundle/Tests/Security/UserAuthenticatorTest.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+use Symfony\Component\Security\Http\Authenticator\AuthenticatorInterface;
+
+class UserAuthenticatorTest extends TestCase
+{
+    public function testThrowsLogicExceptionIfCurrentRequestIsNull()
+    {
+        $firewallMapMock = $this->createMock(FirewallMap::class);
+        $containerInterfaceMock = $this->createMock(ContainerInterface::class);
+
+        $requestStackMock = $this->createMock(RequestStack::class);
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/SecurityBundle/Tests/Security/UserAuthenticatorTest.php]: we don't need a mock here

**Review by nicolas-grekas — COMMENTED** [review]
we prefer using real instances instead of mocks when possible
can you please have a look at your test with this in mind?
(we can use mocks when creating a real instance is too complex of course)

File: src/Symfony/Bundle/SecurityBundle/Tests/Security/UserAuthenticatorTest.php (hunk 1 above)
Comment (fkropfhamer) [reply src/Symfony/Bundle/SecurityBundle/Tests/Security/UserAuthenticatorTest.php]: Did not know that. I'll try to fix!

**Review by chalasr — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @fkropfhamer.

**Pull request merged**